### ISSUE TITLE
InputBox cursor sometimes misbehaves when typing

### ISSUE THREAD

**Issue by tetchel** [eclipse-theia/theia#6886]
### Description
 

Sometimes typing in an InputBox misbehaves. If you type faster, it gets worse and worse. 
You can see [a recording here](https://github.com/eclipse/codewind/files/3936522/Screen.Recording.2019-12-08.at.11.09.16.AM.mov.zip) from this issue: https://github.com/eclipse/codewind/issues/1424

I **only** see this issue with InputBoxes created using `window.createInputBox`. That is, if you use `window.showInputBox`, there is no problem. I see it with InputBoxes that set the `vaildationMessage`, and InputBoxes that don't, so it's not specific to that.

### Reproduction Steps

1. Create an input box using `window.createInputBox`
2. Type in it, really quickly for more obvious results.

**OS and Theia version:** 
Reproduced in both Firefox and Chrome on MacOS.
![image](https://user-images.githubusercontent.com/8888438/72378013-93e32d00-36de-11ea-9d4f-3b8933326816.png)

**Diagnostics:** 
Nothing interesting in the theia logs or browser logs.

**Comment by akosyakov** [comment]
Master state of Theia is 0.15. Did you try with it as well?

**Comment by tetchel** [comment]
Yes, I can. 
[theia-textfield.mp4.zip](https://github.com/eclipse-theia/theia/files/4074343/theia-textfield.mp4.zip)

Interestingly, I tried reproducing it first with the [VS Code quickinput sample](https://github.com/microsoft/vscode-extension-samples/blob/master/quickinput-sample/src/multiStepInput.ts), and I didn't see the bug there.

**Comment by vince-fugnitto** [comment]
> Yes I can, on latest master
> 
> [Screen recording](https://github.com/eclipse-theia/theia/files/4074343/theia-textfield.mp4.zip)
> 
> Interestingly, I tried reproducing it first with the [VS Code quickinput sample](https://github.com/microsoft/vscode-extension-samples/blob/master/quickinput-sample/src/multiStepInput.ts), and I didn't see the bug there. It may be due to [this logic](https://github.com/microsoft/vscode-extension-samples/blob/master/quickinput-sample/src/multiStepInput.ts#L282) in the sample which appears to prevent multiple parallel validations

I verified with the [quickinput multiStep sample](https://github.com/microsoft/vscode-extension-samples/tree/master/quickinput-sample) you mentioned and could not reproduce the issue (I even removed the parallel validation logic). It may simply be how you are validating your string too eagerly.

Would it possible to provide a reproducible example so we can confirm the issue and provide a fix?

**Comment by tetchel** [comment]
>  It may simply be how you are validating your string too eagerly.

Yes, I am validating it in the `onDidChangeValue` callback. But the flickering issue does not occur in VS Code. 

I will try and find the difference between my plugin and the quickinput sample so I can give you a better example for reproduction.

**Comment by vince-fugnitto** [comment]
> I will try and find the difference between my plugin and the quickinput sample so I can give you a better example for reproduction.

Thank you, I appreciate it! :)
